src/DefaultBuilder/src/StaticWebAssets/StaticWebAssetsFileProvider.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore
+{
+    /// <summary>
+    /// A <see cref="IFileProvider"/> for serving static web assets during development.
+    /// </summary>
+    internal class StaticWebAssetsFileProvider : IFileProvider
+    {
+        private static readonly StringComparison FileSystemBasePathComparisonMode = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ?
+            StringComparison.OrdinalIgnoreCase :
+            StringComparison.Ordinal;
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="StaticWebAssetsFileProvider"/>.
+        /// </summary>
+        /// <param name="pathPrefix">The path prefix under which the files in the <paramref name="contentRoot"/> folder will
+        /// be mapped.</param>
+        /// <param name="contentRoot">The absolute path to the content root associated with the static web assets.</param>
+        public StaticWebAssetsFileProvider(string pathPrefix, string contentRoot)
+        {
+            BasePath = pathPrefix.StartsWith("/") ? pathPrefix : "/" + pathPrefix;
+            InnerProvider = new PhysicalFileProvider(contentRoot);
+        }
+
+        public PhysicalFileProvider InnerProvider { get; }
+
+        public string BasePath { get; }
+
+        /// <inheritdoc />
+        public IDirectoryContents GetDirectoryContents(string subpath)
+        {
+            if (!subpath.StartsWith(BasePath, FileSystemBasePathComparisonMode))
```

Comment:
This should only match on whole segments.
"/foobar" starts with "/foo" but shouldn't match.

Reply:
Updated it to use ` PathString.StartsWithSegments` and added a test for it. Thanks @Tratcher!